common/admin/sql_log.py

```diff
@@ -1,43 +1,113 @@
+import uuid
+
 from django.contrib import admin
 from django.core.handlers.wsgi import WSGIRequest
+from django.db import models
 from django.utils.translation import gettext_lazy as _
 
 from common.admin.forms import SQLLogAdminForm
-from common.models import SQLLog
+from common.admin.forms.sql_log import RequestLogAdminForm, SQLLogAdminInlineForm
+from common.models import RequestLog, SQLLog
+from common.querysets.sql_log import RequestLogQuerySet, SQLLogQuerySet
 
 
 @admin.register(SQLLog)
 class SQLLogAdmin(admin.ModelAdmin):
     form = SQLLogAdminForm
     list_display = [
         "_sql",
-        "path",
-        "request_id",
+        "_path",
+        "_request_id",
         "duration_ns",
         "succeeded",
     ]
     readonly_fields = [
         "sql",
-        "path",
-        "body",
+        "request_log",
         "duration_ns",
         "succeeded",
-        "created",
     ]
     list_filter = [
-        "path",
+        "request_log__path",
         "succeeded",
     ]
     search_fields = [
         "sql",
+        "request_log__path",
+        "request_log__body",
+        "request_log__request_id",
+    ]
+
+    def get_queryset(self, request: WSGIRequest) -> SQLLogQuerySet:
+        return super().get_queryset(request).select_related("request_log")
+
+    @admin.display(description=_("SQL"), ordering="sql")
+    def _sql(self, obj: SQLLog) -> str:
+        length = 97
+        return obj.sql[:length] + ("..." if len(obj.sql) > length else "")
+
+    @admin.display(description=_("Path"), ordering="request_log__path")
+    def _path(self, obj: SQLLog) -> str:
+        return obj.request_log.path
+
+    @admin.display(description=_("Request ID"), ordering="request_log__request_id")
+    def _request_id(self, obj: SQLLog) -> uuid.UUID:
+        return obj.request_log.request_id
+
+    def has_add_permission(self, request: WSGIRequest) -> bool:
+        return False
+
+    def has_change_permission(self, request: WSGIRequest, obj: SQLLog | None = None) -> bool:
+        return False
+
+
+class SQLLogAdminInline(admin.TabularInline):
+    model = SQLLog
+    form = SQLLogAdminInlineForm
+    readonly_fields = ["sql", "duration_ns", "succeeded"]
```

Comment:
I'd rather show `duration_ms` instead of `duration_ns` here and actually in most places (We can still show the `duration_ns` value on the SQLLog detail page)

Reply:
Changed this to ms and improved a lot of the formatting. Added a CSV exported for the future.